### PR TITLE
Doctrine usage, services definition, modern tabs

### PR DESCRIPTION
**Pull request by jolelievre**
Adds documentation about:

- how to define services from your modules
- how to add a tab linked to a modern controller
- how to define and user Doctrine entities

Fixes:
- https://github.com/PrestaShop/docs/issues/278
- https://github.com/PrestaShop/docs/issues/210

### CODE REVIEW

File: src/content/1.7/modules/concepts/doctrine.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+[setup-composer]: {{< ref "/1.7/modules/concepts/services/_index.md#setup-composer" >}}
+
+## Define an Entity
+
+Here is a simple example of a `ProductComment` entity:
+
+```php
+// modules/yourmodule/src/Entity/ProductComment.php
+namespace PrestaShop\Module\YourModule\Entity;
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/doctrine.md]:
```suggestion
namespace YourCompany\Module\YourModule\Entity;
```
We use `PrestaShop\Module` for native modules.
It's maybe better to show something different.

File: src/content/1.7/modules/concepts/doctrine.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+## Using Doctrine
+
+### Saving an Entity
+
+Now that your Entity is managed and that your database structure is up to date you can start creating/update data in your database.
+Here is an example in a legacy controller:
+
+```php
+// modules/yourmodule/controllers/front/PostComment.php
+use PrestaShop\Module\ProductComment\Entity\ProductComment;
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/doctrine.md]: same as previous

File: src/content/1.7/modules/concepts/doctrine.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+You can use this as a base to create your own SQL scripts, but remember **not to modify** PrestaShop core database structure and to
+clean the foreign key statements (unless you are **ABSOLUTELY** sure of what you are doing).
+{{% /notice %}}
+
+## Using Doctrine
+
+### Saving an Entity
+
+Now that your Entity is managed and that your database structure is up to date you can start creating/update data in your database.
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/doctrine.md]:
```suggestion
Now that your Entity is managed and your database structure is up to date you can start creating/updating data in your database.
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I'm in favor of chaning namespace `PrestaShop\Module` into something more generic

File: src/content/1.7/modules/concepts/doctrine.md (hunk 1 above)
Comment (jolelievre) [reply src/content/1.7/modules/concepts/doctrine.md]: Nice catch! Done!

File: src/content/1.7/modules/concepts/doctrine.md (hunk 3 above)
Comment (jolelievre) [reply src/content/1.7/modules/concepts/doctrine.md]: thanks

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**